### PR TITLE
Added propagate asg tags to propagate labels and taints on creation

### PR DESCRIPTION
**Pull request by Skarlso**
### Description

Part of https://github.com/weaveworks/eksctl/issues/4853.

Adds a new config value to propagate tags using an explicit setting.

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/apis/eksctl.io/v1alpha5/assets/schema.json [hunk 1]
```diff
@@ -1319,11 +1319,6 @@
         "desiredCapacity": {
           "type": "integer"
         },
-        "disableASGTagPropagation": {
-          "type": "boolean",
-          "description": "disable the tag propagation in case desired capacity is 0.",
-          "x-intellij-html-description": "disable the tag propagation in case desired capacity is 0."
-        },
```
[the remark this answers is not in the document]
Comment (Skarlso) [reply pkg/apis/eksctl.io/v1alpha5/assets/schema.json:1326]: It's pretty recent, and I'm 90% sure no-one used this yet... it's a bit obscure to be fair.

[the remark this answers is not in the document]
Comment (Skarlso) [reply pkg/apis/eksctl.io/v1alpha5/assets/schema.json:1326]: Never the less, you might be right... It has been around for about a month now, I think? Maybe?

[the remark this answers is not in the document]
Comment (aclevername) [reply pkg/apis/eksctl.io/v1alpha5/assets/schema.json:1326]: i think its worth doing something to acknowledge that we no longer read this value. We can deprecate this, and still introduce the new field?

[the remark this answers is not in the document]
Comment (Skarlso) [reply pkg/apis/eksctl.io/v1alpha5/assets/schema.json:1326]: Sure, but I think it's unnecessary really. I don't think anyone even knows about this field. :D

[the remark this answers is not in the document]
Comment (aclevername) [reply pkg/apis/eksctl.io/v1alpha5/assets/schema.json:1326]: mhmmm but what if someone is :see_no_evil:

[the remark this answers is not in the document]
Comment (Skarlso) [reply pkg/apis/eksctl.io/v1alpha5/assets/schema.json]: DONE :D

File: pkg/apis/eksctl.io/v1alpha5/validation.go [hunk 2]
```diff
@@ -112,6 +112,9 @@ func ValidateClusterConfig(cfg *ClusterConfig) error {
 		if err := validateNg(ng.NodeGroupBase, path); err != nil {
 			return err
 		}
+		if ng.DisableASGTagPropagation != nil {
+			logger.Warning("field DisableASGTagPropagation for nodegroup has been deprecated and has no effect. Please use PropagateASGTags instead for nodegroup %s!", ng.Name)
+		}
```
Comment (aclevername) [inline pkg/apis/eksctl.io/v1alpha5/validation.go:117]: do we have a follow up issue to remove the deprecated field? We can be quicker to remove it than the normal 6 months if you like, given very few ppl will have used it

**Review by aclevername — APPROVED** [review]

File: userdocs/src/usage/autoscaling.md [hunk 3]
```diff
@@ -44,10 +44,7 @@ nodeGroups:
       k8s.io/cluster-autoscaler/node-template/taint/feaster: "true:NoSchedule"
 ```
 
-For unmanaged noderoups, this is done by `eksctl` automatically if `desiredCapacity` is set to `0`, there is no need to
-specify these by hand. This feature is not supported for managed nodegroups. However, if the number of tags added like 
-this exceeds the current ASG tag limit ( 50 at the time of this writing ) or, this feature is not desired, there is a
-setting to disable it. Set `disableASGTagPropagation` to `true` like this:
+For unmanaged noderoups, this is done by `eksctl` automatically if `propagateASGTags` to `true` like this:
```
Comment (nikimanoledaki) [inline userdocs/src/usage/autoscaling.md:47]:
```suggestion
For unmanaged noderoups, this is done by `eksctl` automatically if `propagateASGTags` is set to `true` like this:
```

**Review by nikimanoledaki — APPROVED** [review]
What Jake said + LGTM! :)

File: pkg/apis/eksctl.io/v1alpha5/validation.go (hunk 2 above)
Comment (Skarlso) [reply pkg/apis/eksctl.io/v1alpha5/validation.go:117]: https://github.com/weaveworks/eksctl/issues/5007

**Pull request merged**